src/Futhark/CodeGen/Backends/CCUDA.hs

```diff
@@ -326,7 +351,8 @@ callKernel (LaunchKernel safety kernel_name args num_blocks block_size) = do
       grid[perm[1]] = $exp:grid_y;
       grid[perm[2]] = $exp:grid_z;
 
-      void *$id:args_arr[] = { $inits:args'' };
+      int device_id = 0; 
+      void *$id:args_arr[] = { $inits:args'' , &device_id };
```

Comment:
I would prefer if `device_id` was the first argument, if possible.  (So that all the arguments that are _always_ passed come first.)